src/components/WorkPoolCreateForm.vue

```diff
@@ -67,18 +69,23 @@
     return workersCollectionItems.value?.find((item) => item.type === type.value)?.defaultBaseJobConfiguration ?? {}
   })
 
-  const schema = computed<Schema>(() => mapper.map('SchemaResponse', baseJobConfigs.value.variables ?? {}, 'Schema'))
+  const schema = computed<Schema>(() => mapper.map('SchemaResponse', getSchemaWithoutDefaults(baseJobConfigs.value.variables ?? {}), 'Schema'))
+  const schemaDefaultValues = computed(() => getSchemaDefaults(baseJobConfigs.value.variables ?? {}))
   const parameters = ref()
 
-  const testSchema = computed<Schema>(() => {
-    return { ...schema.value, type: 'object' }
+  // Update default values for form selected type changes
+  watch(type, (newType, oldType) => {
+    if (newType !== oldType) {
+      parameters.value = schemaDefaultValues.value
+    }
```

Comment:
This is so the correct values are used when the schema changes yeah? I did something similar for automations when picking a deployment (changing the deployment changes the schema). I added the ability to remember previous values if you switched back and forth. Might be worth implementing that pattern here. 

https://github.com/PrefectHQ/nebula-ui/blob/main/src/components/AutomationActionRunDeployment.vue#L32

Reply:
Oh yeah, that's a much cleaner pattern. Thanks for the tip!